tests/beaker_tests/netdev_stdlib/network_trunk_provider_defaults.rb

```diff
@@ -0,0 +1,142 @@
+# TestCase Name:
+# -------------
+# network_trunk-Provider-Defaults.rb
+#
+# TestCase Prerequisites:
+# -----------------------
+# This is a Puppet ROUTEDINTF resource testcase for Puppet Agent on Nexus devices.
+# The test case assumes the following prerequisites are already satisfied:
+# A. Populating the HOSTS configuration file with the agent and master
+# information.
+# B. Enabling SSH connection prerequisites on the N9K switch based Agent.
+# C. Starting of Puppet master server on master.
+# D. Sending to and signing of Puppet agent certificate request on master.
+#
+# TestCase:
+# ---------
+# This is a ROUTEDINTF resource test that tests for default values for
+# network_trunk resource when created with 'ensure' => 'present'.
+#
+# There are 2 sections to the testcase: Setup, group of teststeps.
+# The 1st step is the Setup teststep that cleans up the switch state.
+# Steps 2-4 deal with network_trunk resource creation and its
+# verification using Puppet Agent and the switch running-config.
+# Steps 5-7 deal with network_trunk resource deletion and its
+# verification using Puppet Agent and the switch running-config.
+#
+# The testcode checks for exit_codes from Puppet Agent, Vegas shell and
+# Bash shell command executions. For Vegas shell and Bash shell command
+# string executions, this is the exit_code convention:
+# 0 - successful command execution, > 0 - failed command execution.
+# For Puppet Agent command string executions, this is the exit_code convention:
+# 0 - no changes have occurred, 1 - errors have occurred,
+# 2 - changes have occurred, 4 - failures have occurred and
+# 6 - changes and failures have occurred.
+# 0 is the default exit_code checked in Beaker::DSL::Helpers::on() method.
+# The testcode also uses RegExp pattern matching on stdout or output IO
+# instance attributes of Result object from on() method invocation.
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+require File.expand_path('../network_trunklib.rb', __FILE__)
+
+result = 'PASS'
+testheader = 'ROUTEDINTF Resource :: All Attributes Defaults'
+
+# @test_name [TestCase] Executes defaults testcase for ROUTEDINTF Resource.
+test_name "TestCase :: #{testheader}" do
+  # @step [Step] Sets up switch for provider test.
+  step 'TestStep :: Setup switch for provider test' do
+    # Define PUPPETMASTER_MANIFESTPATH constant using puppet config cmd.
+    UtilityLib.set_manifest_path(master, self)
+
+    # Expected exit_code is 0 since this is a bash shell cmd.
+    on(master, NetworkTrunkLib.create_defaults)
+
+    # Expected exit_code is 0 since this is a puppet agent cmd with no change.
+    # Or expected exit_code is 2 since this is a puppet agent cmd with change.
+    cmd_str = UtilityLib.get_namespace_cmd(agent, UtilityLib::PUPPET_BINPATH +
+      'agent -t', options)
+    on(agent, cmd_str, acceptable_exit_codes: [0, 2])
```

Comment:
If you add the cleanup step above then the only acceptable_exit_code in this case would be 2.